frontend/js/chatHandler.js

```diff
@@ -0,0 +1,1228 @@
+    this.currentFilter = 'all';
+    this.boundSearchInputHandler = this.searchInputHandler.bind(this);
+  }
+
+  
+  
+  async init(params, router, context) {
+
+    this.router = router;
+
+    // console.log('Params:', params);
+    const url = `wss://${window.location.host}/ws/live_chat/}`;
+    // console.log('WebSocket URL:', url);
+
+    if (this.ws) {
+        console.log('Closing existing WebSocket connection');
+        this.ws.close();
+    }
+
+    console.log('Creating new WebSocket connection');
+    this.ws = new WebSocket(url);
+    
+    this.ws.onopen = () => {
+      console.log('WebSocket connection opened in context:', context);
+      this.ws.send(JSON.stringify({ "type": context, "context": 'setup' }));
+      if (context === 'chat') {
+        this.chatWindowOpened = false;
+      }
+    };
+    
+    this.ws.onerror = (e) => {
+      // console.error('WebSocket error:', e);
+    };
+    
+    this.ws.onmessage = this.onMessage.bind(this);
+    this.ws.onclose = (e) => this.onClose(e, context);
+    
+    if (context === 'chat') {
+      this.currentReceiverId = params.recipient;
+      this.chatWindowOpened = false;
+    } else {
+      this.currentReceiverId = null;
+      // this.initScrollHandling();
+      this.initFiltering();
+      this.initTabHandling();
+      document.getElementById("back").addEventListener("click", async (e) => {
+        router.navigate("/main_menu");
+      });
+      const searchInput = document.getElementById('search-input');
+      searchInput.removeEventListener('keydown', this.boundSearchInputHandler);
+      searchInput.addEventListener('keydown', this.boundSearchInputHandler);
+      }
+    }
+    
+    // async onClose(event) {
+      //   console.log('WebSocket connection closed:', event.code, event.reason);
+      // }
+      
+    async getUserIdfromName(name) {
+      try {
+        const response = await api.get(`/users/${name}`);
+        const json = await response.json();
+        if (response.status === 200) {
+          if (json.user_id) {
+            return json.user_id;
+          }
+          throw new Error('User ID not found');
+        }
+      } catch (error) {
+        console.error('Error fetching user ID:', error);
+        return null;
+      }
+    }
+      
+  async checkToken() {
+    try {
+      const response = await api.get('/token/check');
+      const json = await response.json();
+      if (json['logged-in'] === false) {
+        throw new Error('User is not logged in');
+      }
+      return true;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  async reconnect() {
+    try {
+      this.ws = new WebSocket(`wss://${window.location.host}/ws/live_chat/`);
+    } catch (error) {
+      console.error('Failed to reconnect:', error);
+    }
+  }
+  
+  async onClose(event, context) {
+    console.log('WebSocket connection closed:', event.code, event.reason);
+    
+    if (event.code === 1006) {
+      try {
+        await this.checkToken();
```

Comment:
Why is this exactly needed? What would be the issue if the token is not checked?

Reply:
Also, what is event code 1006?